File: data/code_pairs/fortran/781.f90
program nutest
    implicit none
    integer :: ios
    logical :: l

    open(unit=10, file="foo_open_negative_unit_1.txt", status="replace", iostat=ios)
    if (ios /= 0) then
        print *, "Error opening file foo_open_negative_unit_1.txt"
        stop
    endif
    close(10, status="delete")

    inquire(file="foo_open_negative_unit_1.txt", exist=l)
    if (l) then
        print *, "File foo_open_negative_unit_1.txt should have been deleted."
        stop
    endif

    print *, "Fortran program operations completed."
end program nutest

File: data/code_pairs/cpp/781.cpp
#include <fstream>
#include <iostream>
#include <experimental/filesystem>

namespace fs = std::experimental::filesystem;

int main() {
    std::ofstream fileStream("foo_open_negative_unit_1.txt");
    if (!fileStream) {
        std::cerr << "Error opening file foo_open_negative_unit_1.txt\n";
        std::abort();
    }
    fileStream.close();

    if (!fs::remove("foo_open_negative_unit_1.txt")) {
        std::cerr << "Error deleting file foo_open_negative_unit_1.txt\n";
        std::abort();
    }

    if (fs::exists("foo_open_negative_unit_1.txt")) {
        std::cerr << "File foo_open_negative_unit_1.txt should have been deleted.\n";
        std::abort();
    }

    std::cout << "C++ program operations completed.\n";

    return 0;
}